nisystemlink/clients/systemsstate/models/_state.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+from nisystemlink.clients.systemsstate.models._architecture_enum import Architecture
+from nisystemlink.clients.systemsstate.models._distribution_enum import Distribution
+from nisystemlink.clients.systemsstate.models._feed import Feed
+from nisystemlink.clients.systemsstate.models._package import Package
+from nisystemlink.clients.systemsstate.models._system_image import SystemImage
+
+
+class State(JsonModel):
+    """Basic State entities that can be extended further"""
+
+    name: Optional[str] = None
+    """Gets or sets the name of the state."""
+
+    description: Optional[str] = None
+    """Gets or sets the description of the state."""
+
+    distribution: Distribution
+    """Gets or sets supported distribution by a state."""
+
+    architecture: Architecture
+    "Gets or sets supported architecture by a state."
+
+    properties: Optional[Dict[str, Optional[str]]] = None
+    """Gets or sets the custom properties for a state."""
+
+    workspace: Optional[str]
+    """Gets or sets the ID of the workspace.
+
+    This property is available starting with version 3 of the getStates operation.
+    """
+
+
+class StateMetaData(State):
+    """Model for state metadata information."""
+
+    id: Optional[str] = None
+    """Gets or sets the ID of the state."""
+
+    createdTimestamp: datetime
```

Comment:
For all model fields, use snake_case and let the `JsonModel` convert it to camelCase at runtime serialization.

```suggestion
    created_timestamp: datetime
```

camelCase serialization defined at https://github.com/ni/nisystemlink-clients-python/blob/caddc7501b240b8f4653447b7851f40f3c21de78/nisystemlink/clients/core/_uplink/_json_model.py#L4